src/leader_election.rs

```diff
@@ -238,7 +235,10 @@ pub mod ballot_leader_election {
                     self.majority_connected = false;
                 }
 
-                info!(self.logger, "New Leader elected, pid: {}", top_pid);
+                debug!(
```

Comment:
move `self.hb_round += 1` to above here instead before the logging. Then we don't have to do the confusing and possibly dangerous thing of logging something that is not actually the value with +1.